java/spring/security/audit/nonconstant-location-header.yaml

```diff
@@ -0,0 +1,30 @@
+rules:
+- id: nonconstant-location-header
+  message: Could not resolve the Location header. This is a possible open redirect.
+    If user input can reach this line, an attacker could trick a user to redirect
+    them to  a malicious site. Audit this finding to make sure no user input can reach
+    here, or limit the URLs which can be redirected to using an allowlist.
+  severity: WARNING
+  languages:
+  - java
+  patterns:
+  - pattern-either:
+    - pattern: $SOMETHING.put("$LOCATION", ...)
```

Comment:
can we use type checking to ensure this is `org.springframework.http.ResponseEntity`?  The `put(...)` method is likely to be very noisy on customer code.